drivers/wireless/ieee80211/bcm43xxx/bcmf_netdev.c

```diff
@@ -636,20 +637,54 @@ void bcmf_netdev_notify_rx(FAR struct bcmf_dev_s *priv)
 static int bcmf_ifup(FAR struct net_driver_s *dev)
 {
   FAR struct bcmf_dev_s *priv = (FAR struct bcmf_dev_s *)dev->d_private;
+  irqstate_t flags;
+  uint32_t out_len;
+  int ret;
 
-#ifdef CONFIG_NET_IPv4
-  ninfo("Bringing up: %d.%d.%d.%d\n",
-        (int)(dev->d_ipaddr & 0xff),
-        (int)((dev->d_ipaddr >> 8) & 0xff),
-        (int)((dev->d_ipaddr >> 16) & 0xff),
-        (int)(dev->d_ipaddr >> 24));
-#endif
-#ifdef CONFIG_NET_IPv6
-  ninfo("Bringing up: %04x:%04x:%04x:%04x:%04x:%04x:%04x:%04x\n",
-        dev->d_ipv6addr[0], dev->d_ipv6addr[1], dev->d_ipv6addr[2],
-        dev->d_ipv6addr[3], dev->d_ipv6addr[4], dev->d_ipv6addr[5],
-        dev->d_ipv6addr[6], dev->d_ipv6addr[7]);
-#endif
+  /* Disable the hardware interrupt */
+
+  flags = enter_critical_section();
+
+  if (priv->bc_bifup)
+    {
+      goto errout_in_critical_section;
+    }
+
+  ret = bcmf_wl_active(priv, true);
+  if (ret == OK)
+    {
+      /* Enable chip */
+
+      ret = bcmf_wl_enable(priv, true);
+    }
+
+  if (ret != OK)
+    {
+      goto errout_in_critical_section;
```

Comment:
Do we need to call `bcmf_wl_active(priv, false);` here?

Reply:
Done